evcxr/src/cargo_metadata.rs

```diff
@@ -66,12 +66,21 @@ pub(crate) fn validate_dep(dep: &str, dep_config: &str, config: &Config) -> Resu
         let primary_error_pattern = PRIMARY_ERROR_PATTERN
             .get_or_init(|| Regex::new("(.*) as a dependency of package `[^`]*`").unwrap());
         let mut message = Vec::new();
+        let mut suggest_offline_mode = false;
         for line in String::from_utf8_lossy(&output.stderr).lines() {
             if let Some(captures) = primary_error_pattern.captures(line) {
                 message.push(captures[1].to_string());
             } else if !ignored_lines_pattern.is_match(line) {
                 message.push(line.to_owned());
             }
+
+            if line.contains("failed to fetch `https://github.com/rust-lang/crates.io-index`") {
```

Comment:
Is checking the entire URL necessary?  Is `failed to fetch  ``https://` enough? asking because it's possible to repoint cargo to use crates index hosted somewhere else ...

Reply:
Checking part could trigger on using a non-existing git dependency too. So it depends on how much false positives are acceptable.

Reply:
Fair enough..  repointed crates index is an edge use case anyway (hopefully)